src/state.rs

```diff
@@ -108,20 +165,103 @@ impl State {
 #[cfg(test)]
 mod test {
     use super::*;
+    use crate::piece::Piece;
 
     fn test_board() -> Board {
         Board::initial()
     }
 
+    fn test_simple_board_for_piece_lateral_king(piece: Piece) -> Board {
+        // White king at d4, black king at b7, and white's variable
+        // piece at c4. E.g., the queen on this board:
+        // https://lichess.org/analysis/standard/8/1k6/8/8/2QK4/8/8/8_w_-_-
+        let inner = vec![
```

Comment:
Feature request: parsing a board from a Unicode raw text string.

Reply:
eventually i would like to parse https://en.wikipedia.org/wiki/Forsyth–Edwards_Notation